### PR TITLE
get chat encryption and decryption to work via FTL

### PR DESCRIPTION
**Pull request by maxtaco**
Merge is blocked on #13544

### CODE REVIEW

File: go/chat/teams.go [hunk 1]
```diff
@@ -237,6 +306,11 @@ func (t *TeamsNameInfoSource) DecryptionKey(ctx context.Context, name string, te
 	keyGeneration int, kbfsEncrypted bool) (res types.CryptKey, err error) {
 	defer t.Trace(ctx, func() error { return err },
 		fmt.Sprintf("DecryptionKeys(%s,%s,%v,%d,%v)", name, teamID, public, keyGeneration, kbfsEncrypted))()
+
+	if !kbfsEncrypted && !public && membersType == chat1.ConversationMembersType_TEAM {
```
Comment (mmaxim) [inline go/chat/teams.go]: Don't need to care about KBFS encrypted here

**Comment by maxtaco** [comment]
Needed further fixes:

1. plumb through gregor-based refreshsers
1. allow server to disable FTL if we have an issue.

File: go/chat/teams.go [hunk 2]
```diff
@@ -235,6 +310,17 @@ func (t *TeamsNameInfoSource) EncryptionKey(ctx context.Context, name string, te
 	membersType chat1.ConversationMembersType, public bool) (res types.CryptKey, ni *types.NameInfo, err error) {
 	defer t.Trace(ctx, func() error { return err },
 		fmt.Sprintf("EncryptionKeys(%s,%s,%v)", name, teamID, public))()
+
+	if !public && membersType == chat1.ConversationMembersType_TEAM {
+		m := libkb.NewMetaContext(ctx, t.G().ExternalG())
+		res, ni, err = encryptionKeyViaFTL(m, name, teamID)
+		if retryOnErrorViaFTL(m, err) {
+			err = nil
```
Comment (mmaxim) [inline go/chat/teams.go]: Might be nice to put a comment in here to describe what is going on

File: go/chat/teams.go [hunk 3]
```diff
@@ -257,6 +343,17 @@ func (t *TeamsNameInfoSource) DecryptionKey(ctx context.Context, name string, te
 	keyGeneration int, kbfsEncrypted bool) (res types.CryptKey, err error) {
 	defer t.Trace(ctx, func() error { return err },
 		fmt.Sprintf("DecryptionKeys(%s,%s,%v,%d,%v)", name, teamID, public, keyGeneration, kbfsEncrypted))()
+
+	if !kbfsEncrypted && !public && membersType == chat1.ConversationMembersType_TEAM {
+		m := libkb.NewMetaContext(ctx, t.G().ExternalG())
+		res, err = decryptionKeyViaFTL(m, name, teamID, keyGeneration)
+		if retryOnErrorViaFTL(m, err) {
+			err = nil
```
Comment (mmaxim) [inline go/chat/teams.go]: Same here

**Comment by mmaxim** [comment]
Not sure I follow the offline comment, but I think we discussed in office. Let me know if it is still relevant.

**Comment by maxtaco** [comment]
I tested offline, it works.

**Comment by maxtaco** [comment]
Ok, ready to go once @mlsteele can weigh in on the auditor.

**Comment by mmaxim** [comment]
K this whole thing should be not really in play for anything online/offline related, so glad it checks out.

**Pull request merged**